en/ros2/px4_sdk.md

```diff
@@ -0,0 +1,336 @@
+   [DEBUG] [example_mode_manual]: Publisher found, continuing
+   [DEBUG] [example_mode_manual]: Registering 'My Manual Mode' (arming check: 1, mode: 1, mode executor: 0)
+   [DEBUG] [example_mode_manual]: Subscriber found, continuing
+   [DEBUG] [example_mode_manual]: Publisher found, continuing
+   [DEBUG] [example_mode_manual]: Got RegisterExtComponentReply
+   [DEBUG] [example_mode_manual]: Arming check request (id=1, only printed once)
+   ```
+
+8. On the PX4 shell, you can check that PX4 registered the new mode:
+
+   ```shell
+   commander status
+   ```
+
+   The output should contain:
+
+   ```{5}
+   INFO  [commander] Disarmed
+   INFO  [commander] navigation mode: Position
+   INFO  [commander] user intended navigation mode: Position
+   INFO  [commander] in failsafe: no
+   INFO  [commander] External Mode 1: nav_state: 23, name: My Manual Mode
+   ```
+
+9. At this point you should be able to see the mode in QGroundControl as well:
+
+   ![QGC Modes](../../assets/middleware/ros2/px4_sdk/qgc_modes.png)
+
+10. Select the mode, make sure you have a manual control source (physical or virtual joystick), and arm the vehicle.
+    The mode will then activate, and it should print the following output:
+
+    ```
+    [DEBUG] [example_mode_manual]: Mode 'My Manual Mode' activated
+    ```
+
+11. Now you are ready to create your own mode.
+
+## Mode Tutorial
+
+This section steps through an example for a custom mode class.
+For a complete application, check out the [examples in the repository](https://github.com/PX4/px4_sdk/tree/main/examples/cpp).
+
+```cpp{1,5,7-9,24-31}
+class MyMode : public px4_sdk::ModeBase // [1]
+{
+public:
+  explicit MyMode(rclcpp::Node & node)
+  : ModeBase(node, Settings{"My Mode"}) // [2]
+  {
+    // [3]
+    _manual_control_input = std::make_shared<px4_sdk::ManualControlInput>(*this);
+    _rates_setpoint = std::make_shared<px4_sdk::RatesSetpointType>(*this);
+  }
+
+  void onActivate() override
+  {
+    // Called whenever our mode gets selected
+  }
+
+  void onDeactivate() override
+  {
+    // Called when our mode gets deactivated
+  }
+
+  void updateSetpoint(float dt_s) override
+  {
+    // [4]
+    Eigen::Vector3f thrust_sp{};
+    thrust_sp(2) = -_manual_control_input->throttle();
+    Eigen::Vector3f rates_sp{};
+    rates_sp(0) = _manual_control_input->roll() * 150.F * M_PI / 180.F;
+    rates_sp(1) = -_manual_control_input->pitch() * 150.F * M_PI / 180.F;
+    rates_sp(2) = _manual_control_input->yaw() * 100.F * M_PI / 180.F;
+    _rates_setpoint->update(rates_sp, thrust_sp);
+  }
+
+private:
+  std::shared_ptr<px4_sdk::ManualControlInput> _manual_control_input;
+  std::shared_ptr<px4_sdk::RatesSetpointType> _rates_setpoint;
+};
+```
+
+- **[1]**: First we create a class that inherits from `px4_sdk::ModeBase`
+- **[2]**: In the constructor, we pass the mode name. This also allows to configure some other things, like replacing an FMU-internal mode.
```

Comment:
Can we link to where you show a list of all the things you can configure, or (not quite as good) the instructions on how to override an internal mode.
```suggestion
- **[2]**: In the constructor, we pass the mode name. This also allows us to configure some other things, like replacing an FMU-internal mode.
```